src/archivefiletree.cpp

```diff
@@ -159,7 +147,7 @@ class ArchiveFileTreeImpl: public virtual ArchiveFileTree, public virtual Archiv
     return std::make_shared<ArchiveFileTreeImpl>(parent, name, -1, std::vector<File>{});
   }
 
-  virtual void doPopulate(std::shared_ptr<const IFileTree> parent, std::vector<std::shared_ptr<FileTreeEntry>>& entries) const override {
+  virtual bool doPopulate(std::shared_ptr<const IFileTree> parent, std::vector<std::shared_ptr<FileTreeEntry>>& entries) const override {
```

Comment:
What does the bool signify?

Reply:
It's explained in `IFileTree` (this is an overiden method) - It says if the vector of entries has been sorted or not (and thus should be sorted by the parent class):

```cpp
/**
 * @brief Method that child classes should implement.
 *
 * This method should populate the given entries of this tree by using the makeFile and makeTree method. The
 * usage of the makeFile and makeTree method is not mandatory here.
 *
 * If the implementation can populate the vector of entries in order, it is possible to return false to
 * tell IFileTree not to re-sort the vector. If sorted, directories should be before files, and both directories
 * and files should be sorted by name in a case-insensitive way.
 *
 * @param parent The current tree, without const-qualification.
 * @param entries Vector of entries to populate.
 *
 * @return true if the vector of entries is already sorted, false if it must be sorted.
 */
```

Reply:
Yeah sorry I found it a little later in the uibase pr, thanks.

Reply:
It's for cases where you can sort faster in the child method. E.g., for `QDirFileTree`, it seems to be a tad faster to sort using `QDir::SortFlags` with `entryInfoList` that sorting at the end.